### DISCUSSION TITLE
Amiga/BlueSCSIv2: Testing Results

### DISCUSSION THREAD

**Discussion by salocinx** [BlueSCSI/BlueSCSI-v2#136] [General]
Hi. I have been testing the BlueSCSI v2 on various Amiga SCSI controllers.

Device used for testing: Blue SCSI v2 => HW: 2023.10a, FW: BlueSCSI_Pico_2024-02-22_ceb4b8e1.uf2

**[Amiga 3000's Internal SCSI Bus](http://amiga.resource.cx/mod/a3000.html)**

As already stated in the [compatibility list](https://bluescsi.com/docs/Compatibility), I herewith can confirm that the BlueSCSI v2 works fine in the Amiga 3000 desktop model by using the suggested bluescsi.ini settings:

[SCSI]
System="Generic"
EnableSCSI2=0 ; Off

**[Commodore A4091](http://amiga.resource.cx/exp/a4091)**

Notes: Works perfectly fine with the default bluescsi.ini (SCSI2 enabled).

[SCSI]
System="Generic"
EnableSCSI2=1 ; On

Transfer Rate: between 6000-8000 kB/s (6-8MB/s)

Notes: If you use the A4091 in an Amiga 3000 desktop model and you do not have any devices connected to the Amiga 3000's own internal SCSI bus, it's very important that you nevertheless properly terminate the Amiga 3000's internal SCSI bus (with internal termination resistor packs and an external SCSI terminator). Otherwise the Amiga 3000's internal SCSI bus will mess up with DMA on Super-DMAC.

**[Commodore A590](http://amiga.resource.cx/exp/a590)** on Amiga 500

Settings: Switch 1 (Auto Boot) = ON, Switch 2 (LUN) = ON, Switch 3 (Time-Out) = OFF

bluescsi.ini:

[SCSI]
System="Generic"
EnableSCSI2=0 ; Off
PrefetchBytes=0

Transfer Rate: around 900 kB/s

**[GVP A530](http://amiga.resource.cx/exp/gvp530)** on Amiga 500

Settings: Do not forget to set jumper **CN18** from the default 1-2 to **2-3**. This will enable the 14 MHz mode (default 7 MHz) !

bluescsi.ini:

[SCSI]
System="Generic"
EnableSCSI2=0 ; Off
PrefetchBytes=0

Transfer Rate: around 2080 kB/s

Notes: Tested with GVP Boot ROM v4.15 and GuruROM v6.14, both basically work fine until an external CD-ROM is attached to the A530 external SCSI port (does not boot anymore). Will further investigate the issue, probably not BlueSCSI related.

**[GVP A2000-HC Rev 3](http://amiga.resource.cx/exp/impact2000hc)**

bluescsi.ini:

[SCSI]
System="Generic"
EnableSCSI2=0 ; Off
PrefetchBytes=0

Transfer Rate: around 900 kB/s

Notes: In case an external SCSI devices is attached to the external SCSI port (e.g. CD-ROM), the termination jumper had to be disabled (TERM=OFF) in order to make the system work. If no external SCSI device is attached, enable termination on the BlueSCSIv2 device (TERM=ON). Otherwise you might get the following error while booting the operating system:

```
gvpscsi.device: unit 001
Unexpected Status $4E/$20
```

**[GVP A2000-HC+8 Series II Rev 2](http://amiga.resource.cx/exp/impact2000hc8)**

Boot ROM: GVP ROM v4.15

bluescsi.ini:

[SCSI]
System="Generic"
EnableSCSI2=0 ; Off
PrefetchBytes=0

Transfer Rate: around 1200 kB/s

Notes: In case an external SCSI devices is attached to the external SCSI port (e.g. CD-ROM), the termination jumper had to be disabled (TERM=OFF) in order to make the system work. If no external SCSI device is attached, enable termination on the BlueSCSIv2 device (TERM=ON). Otherwise you might get the following error while booting the operating system:

```
gvpscsi.device: unit 001
Unexpected Status $4E/$20
```

**[Commodore A2091](http://amiga.resource.cx/exp/a2091)**

Boot ROM: Commodore v7.0
SCSI Controller Chip: WD33C93A-PL 00-04

bluescsi.ini:

[SCSI]
System="Generic"
EnableSCSI2=0 ; Off
PrefetchBytes=0

Transfer Rate: only around 150 kB/s

Notes: Need to further investigate the low transfer rate. The A2091 can be modified to work in 14 MHz. Here's a good [tutorial to upgrade the A2091](http://blog.worldofjani.com/?p=7485). It should be possible to get 1.6-2.8 MB/s transfer rates, I will try and report back.

**Comment by scalarmapedia** [comment]
Carumba, thats a LOT of testing. How do you have ALL of these devices?

**Comment by salocinx** [comment]
I have even more : ) I am Amiga enthusiast, collector and interested in new developments. I will post more test results in the future.

**Comment by scalarmapedia** [comment]
Hi, I have a Fastland Z3 SCSI card for Amiga 3000/4000, wondering if you could offer any advice on it. As you have already tested, and got working so many Amiga SCSI devices with the BlueSCSI, your experience would be invaluable. Would you be able to help?

**Reply by salocinx** [reply]
Hi. I once had a Fastlane Z3, but sold it and built myself an A4091. But I can ask a friend if he will borrow it to me in order to make some tests in the next 2-3 weeks.

- I am not sure if the Fastlane Z3 needs Super-Buster 11, please check the Internet. The Fastlane Z3 might be able to use FastSCSI2 (using EnableSCSI2=1 and PrefetchBytes=8192 in bluescsi.ini; has to be tested)
- The A3000D is very picky when using any 040/060 CPU accelerator, almost everybody encounters read/write and checksum errors. However, when using the internal 030 CPU, everything works fine (using EnableSCSI2=0 and PrefetchBytes=0 in bluescsi.ini)
- If you are using an 060 CPU accelerator, you will also need the latest v8.5 boot ROM on the Fastlane Z3, that's for sure.
- Probably the most important thing with SCSI is termination: Always make sure you have both ends of the SCSI chain terminated properly. Sometimes there are resistor packs on the boards for this purpose. Hard drives and BlueSCSIv2 have termination jumpers. Other times you need external SCSI termination devices. There is a lot of information in the Internet.

Good luck! I will report back if I can borrow the Fastlane Z3.

**Comment by scalarmapedia** [comment]
_Hi. I once had a Fastlane Z3, but sold it and built myself an A4091. But I can ask a friend if he will borrow it to me in order to make some tests in the next 2-3 weeks._

That would be great. Right now, I can't see ANY others out there using this card. Also, buying a real SCSI drive is not viable, due to high cost of SCSI1 / SCSI2 era devices, I dont want to get a newer device, as that would also require expensive adapters. I DO have a Quantum drive, from an A2000's SCSI card, but I replaced THAT with a SCSI2SD and testing it on my GVP HD8+, it was not accessible.

So far I have tried BlueSCSIv2, ZuluSCSI rp 2040 and last night a SCSI2SD v5.1. They all displayed Checksum errors when opening the Phase5 SCSI app, and then those drives not mounting, or mounting sporadically when the drives were partioned. Then not wanting to format (not a DOS disk) or formatting, but then proving to be unwritable, with Checksum errors.

_I am not sure if the Fastlane Z3 needs Super-Buster 11, please check the Internet._ 

Phase 5 stated in their docs, that it specifically does NOT need the Buster 11. However, it says the card can use less CPU when its installed, but that this MIGHT need a mod on it. Its possible this mod was done, and the previous owner had a Buster 11. I currently have have a Buster 9. I may go and get an 11, as I am a 15 minute drive from Novalogic (main store of them). I can then keep the 9 as a spare.

_The Fastlane Z3 might be able to use FastSCSI2 (using EnableSCSI2=1 and PrefetchBytes=8192 in bluescsi.ini; has to be tested)_

I have enabled both, and it says it can use FastSCS2, but until I can get it to be basically reliable at all. Its not worth dabbling with.

_The A3000D is very picky when using any 040/060 CPU accelerator, almost everybody encounters read/write and checksum errors. However, when using the internal 030 CPU, everything works fine (using EnableSCSI2=0 and PrefetchBytes=0 in bluescsi.ini)_

This is an A4000A with an original 68040 CPU board, rev 3.1, which apparently means its buffer chip does not have to be replaced (in Z3 manual), it does worry me though, but it appears my card and that chip are ok.

_If you are using an 060 CPU accelerator, you will also need the latest v8.5 boot ROM on the Fastlane Z3, that's for sure._

The card has an 8.5 ROM on it, but its a 3.2 revision card, which did not have it, so it looks like prev owner DID upgrade it. This makes me wonder if they also modded it to get the Buster 11 buff....

_Probably the most important thing with SCSI is termination: Always make sure you have both ends of the SCSI chain terminated properly. Sometimes there are resistor packs on the boards for this purpose. Hard drives and BlueSCSIv2 have termination jumpers. Other times you need external SCSI termination devices. There is a lot of information in the Internet._

Currently the card has 3 resistor packs and they are installed on the card. The BlueSCSI uses auto termination, and I have its TerminationON jumper set. So it should work fine. These cards all work fine BTW in my GVP HD8+. I suppose something other than these resistor packs could be iffy, a broken trace to these resistors.

I may park my work on this card for now. If you CAN get another card, it would be ideal, obviously having two examples would really help diagnose things.

Let me know if I can do anything to help in the meantime.

**Comment by scalarmapedia** [comment]
Hey, so I found the issue! Was a bit of a guess in the end, but it worked. Ill be releasing a video on the whole saga in the next few days.

As you know, with the Super Buster 9 in place, the Fastland Z3 manifested numerous subtle issues, mainly in the form of Checksum errors, but also drives would not mount, or mount infrequently and not work.

Basically, although the card does NOT need a Super Buster 11 from Factory (a big feature at the time), because of historical upgrades to the card, it now DOES.

It appears some previous owner had upgraded it, I could see, to the latest 8.5rom, and this meant another upgrade to make the card use the native Buster 11 DMA instead of that on the card was possible. This involves replacing some of the chips on the card. But previous owners had covered ALL the customs ICs with heat sinks, so hard to scrutinize. Anyway, I guessed they HAD done this, which meant the card was now relying on the Buster 11 chip, which I did not have...

Lukily I live 20 minutes away from Analogic in Kingston UK, so I quickly popped there, grabbed an 11 . Popped it in that evening, and immediately a drive which had previously not mounted, DID, I formatted it, no errors, copied files to it, no errors :)

I want to get a graphics card, maybe a zz9000 for my A4000, so I felt that, even if the 11 did NOT fix the problem, it was not a waste of cash. Luckily, it did.

There were many other issues along the way, so I feel i need to document those for anyone else dealing with one of these cards. I see a LOT of these have DEAD SCSI on them. Maybe their not actually dead...

**Reply by salocinx** [reply]
Great, well done : D ! A documentation video would be awesome, also to link in the compatibility list. Let us know when you publish it. Thank you!
